### PR TITLE
Add logic to cancel older running builds in Jenkins

### PR DESCRIPTION
**Pull request by pedropombeiro**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

### Summary:

[comment]: # (Summarise the problem and how the pull request solves it)
This PR addresses https://www.pivotaltracker.com/story/show/163367849.

### Review notes (optional):

Test build canceled by newer build: https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7317/18/

status: ready

### CODE REVIEW

**Review by mandrigin — APPROVED** [review]
Looks good! But the PR title is misleading a bit.

**Review by jakubgs — APPROVED** [review]
That's a lot of whitespace around that call.
Great work.

**Comment by pedropombeiro** [comment]
> That's a lot of whitespace around that call.

I thought it would be good to have a visual marker around the point where builds are canceled, but you're right, it's not really needed.

**Comment by jakubgs** [comment]
Just a nitpick, no worries.

**Comment by pedropombeiro** [comment]
Should the child also try to cancel the upstream job (so that that one cancels the children)?

**Comment by jakubgs** [comment]
Wouldn't that be handled by the previous job? Or are you worrying about some kind of race condition?

File: ci/common.groovy [hunk 1]
```diff
@@ -29,6 +32,25 @@ def getBuildType() {
   return params.BUILD_TYPE
 }
 
+@NonCPS
+def abortPreviousRunningBuilds() {
+  Run previousBuild = currentBuild.rawBuild.getPreviousBuildInProgress()
+
+  while (previousBuild != null) {
```
Comment (jakubgs) [inline ci/common.groovy:39]: I guess there's no harm in iterating and making sure all are killed.

**Comment by pedropombeiro** [comment]
Yeah, ideally it would be handled by the upstream job directly, but I don't see where I'd do that.

**Comment by jakubgs** [comment]
>but I don't see where I'd do that.

But it would have already done that when it started. Meaning, if the newest job is trying to kill the still running previous job, that would mean to me that the job before the previous job(if there ever was one) was already killed by the previous job.

Unless some kind of race condition happened I guess.

**Comment by pedropombeiro** [comment]
Nevermind, I thought this job was similar to the `Jenkinsfile.combined` one, where the child jobs are started in a `parallel` entry, but apparently not. So right now the only delay that exists is if the build queue is too large for the new builds to start executing and start killing running builds, but I guess that can be improved in the future.

**Comment by jakubgs** [comment]
>So right now the only delay that exists is if the build queue is too large for the new builds to start executing and start killing running builds, but I guess that can be improved in the future.

Good point, but this is already a big improvement that should shorten the queues, so this scenario is less likely.

**Pull request merged**